apps/status-api/src/providers/WhaleApiClientProvider.ts

```diff
@@ -0,0 +1,30 @@
+import { WhaleApiClient } from '@defichain/whale-api-client'
+import { Injectable } from '@nestjs/common'
+import { SupportedNetwork } from '../pipes/NetworkValidationPipe'
+
+@Injectable()
+export class WhaleApiClientProvider {
+  private readonly clientCacheByNetwork: Map<SupportedNetwork, WhaleApiClient> = new Map()
```

Comment:
We should follow `ocean-api`, `whale` structure where there is only one NETWORK type per server/instance.

Reply:
So we can let the Load Balancer do the routing via, e.g. `/v0/mainnet` and `/v0/testnet`